src/core/useCubeCamera.tsx

```diff
@@ -0,0 +1,50 @@
+import * as THREE from 'three'
+import { HalfFloatType, Fog, FogExp2, WebGLCubeRenderTarget } from 'three'
+import * as React from 'react'
+import { useMemo } from 'react'
+import { useThree } from '@react-three/fiber'
+
+export type Props = {
+  /** Resolution of the FBO, 256 */
+  resolution?: number
+  /** Camera near, 0.1 */
+  near?: number
+  /** Camera far, 1000 */
+  far?: number
+  /** Custom environment map that is temporarily set as the scenes background */
+  envMap?: THREE.Texture
+  /** Custom fog that is temporarily set as the scenes fog */
+  fog?: Fog | FogExp2
+}
+
+export function useCubeCamera({ resolution = 256, near = 0.1, far = 1000, envMap, fog }: Props = {}) {
+  const gl = useThree(({ gl }) => gl)
+  const scene = useThree(({ scene }) => scene)
+
+  const fbo = useMemo(() => {
+    const fbo = new WebGLCubeRenderTarget(resolution)
+    fbo.texture.encoding = gl.outputEncoding
+    fbo.texture.type = HalfFloatType
+    return fbo
+  }, [resolution])
```

Comment:
This will need to be cleaned up afterwards:

```jsx
useEffect(() => {
  return () => {
    fbo.dispose()
  }
}, [fbo])
```

Reply:
ok, I've added the clean-up effect 👍🏻